onnx/common/path.cc

```diff
@@ -9,99 +9,79 @@
+          }                                                                                                   \
+          out.push_back(dot_char);                                                                            \
+          out.push_back(dot_char);                                                                            \
+          dotdot = out.size();                                                                                \
+        }                                                                                                     \
+        continue;                                                                                             \
+      }                                                                                                       \
+                                                                                                              \
+      if (!out.empty() && out.back() != sep) {                                                                \
+        out.push_back(sep);                                                                                   \
+      }                                                                                                       \
+                                                                                                              \
+      for (; r < n && !is_path_separator(path[r]); r++) {                                                     \
+        out.push_back(path[r]);                                                                               \
+      }                                                                                                       \
+    }                                                                                                         \
+                                                                                                              \
+    if (out.empty()) {                                                                                        \
+      out.push_back(dot_char);                                                                                \
+    }                                                                                                         \
+                                                                                                              \
+    normalize_separator(out);                                                                                 \
+                                                                                                              \
+    return out;                                                                                               \
   }
 
-  // Use 1 separator in path.
-  normalize_separator(out);
+PATH_JOIN(std::string, k_preferred_path_separator);
+CLEAN_RELATIVE_PATH(std::string, char, k_preferred_path_separator, ".");
 
-  return out;
-}
+#ifdef _WIN32
+PATH_JOIN(std::wstring, w_k_preferred_path_separator);
```

Comment:
Please try to not implement such things by yourself. Use the functions in std::filesystem instead. Because it is very very difficult to handle all the cases.  Path names on Windows have many different styles.